src/jaxsim/api/data.py

```diff
@@ -374,11 +378,7 @@ def base_orientation(self, dcm: jtp.BoolLike = False) -> jtp.Vector | jtp.Matrix
         # we introduce a Baumgarte stabilization to let the quaternion converge to
         # a unit quaternion. In this case, it is not guaranteed that the quaternion
         # stored in the state is a unit quaternion.
-        W_Q_B = jax.lax.select(
-            pred=jnp.allclose(jnp.linalg.norm(W_Q_B), 1.0, atol=1e-6, rtol=0.0),
-            on_true=W_Q_B,
-            on_false=W_Q_B / jnp.linalg.norm(W_Q_B),
-        )
+        W_Q_B = W_Q_B / jnp.linalg.norm(W_Q_B)
```

Comment:
For AD purpose, having a division with a square root is suboptimal. I'd try to apply normalization only if needed:

```suggestion
        W_Q_B = jnp.where(
            W_Q_B.dot(W_Q_B) == 1.0, W_Q_B, W_Q_B / jnp.linalg.norm(W_Q_B)
        )
```

Reply:
Thanks! Addressed in https://github.com/ami-iit/jaxsim/pull/280/commits/03edbf2a6a1c6ca63711312b04e349cb02312392

Reply:
Should be use allclose or == 1? Strict comparison with floats are tricky. I think we can support quaternions with a norm within the default tolerances of allclose.

Reply:
Addressed with rebase